relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -307,21 +228,12 @@ mod tests {
     }
 
     #[test]
-    /// Should remove transaction from envelope when a matching rule is detected
-    fn test_should_drop_transaction() {
+    fn test_basic_trace_sampling() {
```

Comment:
nit: a more meaningful name for the test; for example (and maybe too long):
```suggestion
    fn test_unsampled_envelope_with_sample_rate() {
```